src/device/device-management.ts

```diff
@@ -29,38 +36,64 @@ export class WcmeError {
   }
 }
 
-export type AudioDeviceConstraints = {
-  deviceId?: string;
+export type CustomVideoTrackInitConfig = {
+  mediaStreamTrack: MediaStreamTrack;
+  encoderConfig?: VideoEncoderConfig;
 };
 
-export type VideoDeviceConstraints = {
-  deviceId?: ConstrainDOMString;
-  width?: ConstrainULong;
-  height?: ConstrainULong;
-  aspectRatio?: ConstrainDouble;
-  frameRate?: ConstrainDouble;
-  facingMode?: ConstrainDOMString;
+export type CustomAudioTrackInitConfig = {
+  mediaStreamTrack: MediaStreamTrack;
+  encoderConfig?: AudioEncoderConfig;
 };
 
+/**
+ * Creates MicrophoneTrack and CameraTrack at the same time.
+ *
+ * @param audioConstraints - Audio constraints to create microphone track.
+ * @param videoConstraints - Video constraints to create camera track.
+ * @returns MicrophoneTrack and cameraTrack at same time.
+ */
+export async function createMicrophoneAndCameraTracks(
+  audioConstraints?: MicrophoneConstraints,
+  videoConstraints?: VideoConstraints
+): Promise<[LocalMicrophoneTrack, LocalCameraTrack]> {
+  let stream: MediaStream;
+  try {
+    // TODO: change it
```

Comment:
change what? i'd either get rid of the TODO or add more detail

Reply:
done